File: 005pairSum.js
const pairSum = (numbers, targetSum) => {
  // todo
  const result = []
  for (let i = 0; i < numbers.length; i++){
    const oneNum = numbers[i]
    for (let j = 1; j < numbers.length; j++){
      const twoNum = numbers[j]
      if (oneNum + twoNum === targetSum){
        result.push(i, j)
        return result
      }
    }
  }
};
console.log(pairSum([3, 2, 5, 4, 1], 8)); // -> [0, 2]

File: 005pair_sum.py
# brute force solution
# def pair_sum(numbers, target_sum):
#     for i in range(0, len(numbers)):
#         for j in range(i + 1, len(numbers)):
#             if numbers[i] + numbers[j] == target_sum:
#                 return (numbers[i], numbers[j])
#     return None



# optimized solution
def pair_sum(numbers, target_sum):
  previous_numbers = {}

  for index, num in enumerate(numbers):
    complement = target_sum - num

    if complement in previous_numbers:
      return (previous_numbers[complement], index)

    previous_numbers[num] = index
